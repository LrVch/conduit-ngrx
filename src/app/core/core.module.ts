import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
