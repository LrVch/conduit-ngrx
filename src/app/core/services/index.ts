export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
