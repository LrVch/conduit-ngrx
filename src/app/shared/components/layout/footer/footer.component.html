<footer class="footer">
  <mat-toolbar color="primary">
    <div class="container">
      <a routerLink="/"
         class="logo footer__title">conduit NGRX
      </a>
      <span class="footer__attribution"> Â© 2018.
        Code licensed under MIT.
      </span>
    </div>
  </mat-toolbar>
</footer>