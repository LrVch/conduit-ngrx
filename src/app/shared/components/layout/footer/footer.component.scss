.footer {
    position: relative;
    color: #fff;
    a {
        &.logo {
            text-decoration: none;
            @media (max-width: 599px) {
                font-size: 15px;
            }
        }
        color: #fff;
        text-decoration: underline;
        &:hover {
            color: #fff;
            opacity: .8;
        }
    }
    .mat-toolbar-single-row {
        height: auto !important;
        min-height: 64px;
    }
}

a.footer__title {
    margin-right: 20px;
    text-decoration: none;
}

.footer__attribution {
    font-size: 12px;

    @media (max-width: 599px) {
        display: block;
    }
}