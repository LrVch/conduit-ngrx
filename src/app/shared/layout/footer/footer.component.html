<footer class="footer">
  <mat-toolbar color="primary">
    <div class="container">
      <a routerLink="/"
         class="logo footer__title">conduit NGRX
      </a>
      <span class="footer__attribution"> Â© 2018.
        <!-- An interactive learning project from
          <a href="https://thinkster.io">Thinkster</a>. -->
        Code licensed under MIT.
      </span>
    </div>
  </mat-toolbar>
</footer>