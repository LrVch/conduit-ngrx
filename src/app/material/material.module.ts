import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule
];

@NgModule({
  imports: [...modules],
  exports: [...modules]
})
export class MaterialModule { }
