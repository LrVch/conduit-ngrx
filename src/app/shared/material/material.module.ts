import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
