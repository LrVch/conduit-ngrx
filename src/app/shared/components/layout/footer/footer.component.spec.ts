import { async, ComponentFixture } from '@angular/core/testing';
import { ConfigureFn, configureTests } from '@app/lib/testing';
import { FooterComponent } from './footer.component';
import { DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '@app/shared/material/material.module';
import { RouterLinkDirectiveStubDirective } from '@app/lib/testing/directive/router-link-directive-stub';

describe('FooterComponent', () => {
    let component: FooterComponent;
    let fixture: ComponentFixture<FooterComponent>;
    let de: DebugElement;
    let linkDes;
    let routerLinks;

    beforeEach(
        async(() => {
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [
                        FooterComponent,
                        RouterLinkDirectiveStubDirective
                    ],
                    imports: [MaterialModule]
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(FooterComponent);
                component = fixture.componentInstance;
                de = fixture.debugElement;
                fixture.detectChanges();
                linkDes = de.queryAll(By.directive(RouterLinkDirectiveStubDirective));
                routerLinks = linkDes.map(deel => deel.injector.get(RouterLinkDirectiveStubDirective));
            });
        })
    );

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should compile', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('can get RouterLinks from template', () => {
        fixture.detectChanges();

        expect(routerLinks.length).toBe(1, 'should have 1 routerLink');
        expect(routerLinks[0].linkParams).toEqual('/');
    });

    it('can click Heroes link in template', () => {
        const goToDe = linkDes[0];
        const goToLink = routerLinks[0];

        expect(goToLink.navigatedTo).toBeNull();

        fixture.detectChanges();
        goToDe.triggerEventHandler('click', null);

        expect(goToLink.navigatedTo).toEqual('/');
    });
});

