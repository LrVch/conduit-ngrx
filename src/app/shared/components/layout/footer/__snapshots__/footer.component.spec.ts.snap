// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should compile 1`] = `
<app-footer>
  <footer
    class="footer"
  >
    <mat-toolbar
      class="mat-toolbar mat-primary mat-toolbar-single-row"
      color="primary"
      ng-reflect-color="primary"
    >
      <div
        class="container"
      >
        <a
          class="logo footer__title"
          ng-reflect-link-params="/"
          routerlink="/"
        >
          conduit NGRX 
        </a>
        <span
          class="footer__attribution"
        >
           Â© 2018. Code licensed under MIT. 
        </span>
      </div>
    </mat-toolbar>
  </footer>
</app-footer>
`;
